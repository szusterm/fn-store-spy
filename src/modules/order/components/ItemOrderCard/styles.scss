@import '~material-colors/dist/colors';

@import '~scss/_transition';
@import '~scss/_input';
@import '~scss/_opacity-colors';
@import '~scss/_center-items';

.item-order-card {
	@include input-reset();
	position: relative;
	cursor: pointer;

	&:hover, &:focus {
		.item-order-card--action-box {
			opacity: 1;

			.action-box--sign {
				transform: scale(1);
			}
		}
	}

	&--action-box {
		position: absolute;
		top: 0;
		bottom: 0;
		left: 0;
		right: 0;

		border-radius: 50%;
		opacity: 0;

		@include black-background(0.6);
		@include transition(sh);
		@include center-items();

		.action-box {
			&--sign {
				padding: 5px;

				border-radius: 50%;
				background-color: $md-red-500;
				color: white;

				transform: scale(0);

				@include transition(md);
			}
		}
	}
}