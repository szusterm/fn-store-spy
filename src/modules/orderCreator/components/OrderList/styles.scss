.order-list {
	padding: 20px !important;
	max-height: 100%;
	overflow: auto;

	&--item-box {
		padding: 6px !important;
	}
}