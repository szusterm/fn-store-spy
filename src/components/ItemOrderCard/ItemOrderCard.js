import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import {removeItemFromOrderByIndex} from '../../redux/actions';

import ItemCoin from '../ItemCoin';

import './styles.scss';

export class ItemOrderCard extends Component {
	render() {
		const {imageSrc, rarity} = this.props;

		return (
			<div className={'item-order-card'}>
				<ItemCoin
					imageSrc={imageSrc}
					rarity={rarity}
				/>
			</div>
		);
	}
}

ItemOrderCard.propTypes = {
	index: PropTypes.number.isRequired,
	imageSrc: PropTypes.string.isRequired,
	rarity: PropTypes.string.isRequired,
	removeItemFromOrderByIndex: PropTypes.func //redux
};

const mapDispatchToProps = {removeItemFromOrderByIndex};

export default connect(null, mapDispatchToProps)(ItemOrderCard);