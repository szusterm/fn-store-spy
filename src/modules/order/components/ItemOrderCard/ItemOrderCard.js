import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import {removeItemFromOrderByIndex} from '../../../../redux/actions';

import ItemCoin from '../../../../components/ItemCoin/index';

import './styles.scss';

export class ItemOrderCard extends Component {
	constructor(props) {
		super(props);

		this.removeItemFromOrder = this.removeItemFromOrder.bind(this);
	}

	removeItemFromOrder() {
		const {index} = this.props;
		this.props.removeItemFromOrderByIndex(index);
	}

	render() {
		const {imageSrc, rarity} = this.props;

		return (
			<button
				className={'item-order-card'}
				onClick={this.removeItemFromOrder}
			>
				<ItemCoin
					imageSrc={imageSrc}
					rarity={rarity}
				/>
				<div className={'item-order-card--action-box'}>
					<div className={'material-icons action-box--sign'}>
						<span>clear</span>
					</div>
				</div>
			</button>
		);
	}
}

ItemOrderCard.propTypes = {
	index: PropTypes.number.isRequired,
	imageSrc: PropTypes.string.isRequired,
	rarity: PropTypes.string.isRequired,
	removeItemFromOrderByIndex: PropTypes.func //redux
};

const mapDispatchToProps = {removeItemFromOrderByIndex};

export default connect(null, mapDispatchToProps)(ItemOrderCard);