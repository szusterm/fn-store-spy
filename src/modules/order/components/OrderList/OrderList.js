import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import './styles.scss';

export class OrderList extends Component {

}

export default OrderList;