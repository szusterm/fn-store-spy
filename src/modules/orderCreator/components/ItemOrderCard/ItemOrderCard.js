import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import {removeItemFromOrderByIndex} from '../../actions';

import ItemCoin from '../../../../components/ItemCoin';

import './styles.scss';

export class ItemOrderCard extends Component {
	constructor(props) {
		super(props);

		this.removeItemFromOrder = this.removeItemFromOrder.bind(this);
	}

	removeItemFromOrder() {
		const {index} = this.props;
		this.props.removeItemFromOrderByIndex(index);
	}

	render() {
		const {name, imageSrc, rarity} = this.props;

		return (
			<button
				className={'item-order-card'}
				onClick={this.removeItemFromOrder}
			>
				<ItemCoin
					imageSrc={imageSrc}
					rarity={rarity}
					name={name}
				/>
				<div className={'item-order-card--action-box'}>
					<div className={'material-icons action-box--sign'}>
						<span>clear</span>
					</div>
				</div>
			</button>
		);
	}
}

ItemOrderCard.propTypes = {
	index: PropTypes.number.isRequired,
	name: PropTypes.string.isRequired,
	imageSrc: PropTypes.string.isRequired,
	rarity: PropTypes.string.isRequired,
	removeItemFromOrderByIndex: PropTypes.func //redux
};

const mapDispatchToProps = {removeItemFromOrderByIndex};

export default connect(null, mapDispatchToProps)(ItemOrderCard);