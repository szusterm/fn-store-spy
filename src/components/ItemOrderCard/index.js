import ItemOrderCard from './ItemOrderCard';

export default ItemOrderCard;