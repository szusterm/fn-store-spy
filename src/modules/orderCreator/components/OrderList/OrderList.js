import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import ItemOrderCard from '../ItemOrderCard';

import './styles.scss';

export class OrderList extends Component {
	getItemColsWidth() {
		const {items} = this.props;
		return (items.length <= 9) ? 4 : 3;
	}

	render() {
		const {items} = this.props;

		return (
			<div className={'container-fluid order-list'}>
				<div className={'row'}>
					{
						items.map((item, index) => {
							const {_id, imageSrc, rarity} = item;

							return (
								<div
									key={_id}
									className={`col-${this.getItemColsWidth()} order-list--item-box`}
								>
									<ItemOrderCard
										index={index}
										imageSrc={imageSrc}
										rarity={rarity}
									/>
								</div>
							);
						})
					}
				</div>
			</div>
		);
	}
}

OrderList.propTypes = {
	items: PropTypes.array //redux
};

const mapStateToProps = (state) => {
	const {items} = state.orderCreator;
	return {items};
};

export default connect(mapStateToProps, null)(OrderList);