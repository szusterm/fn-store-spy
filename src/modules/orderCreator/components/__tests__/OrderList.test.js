import React from 'react';
import {shallow} from 'enzyme';

import ItemOrderCard from '../ItemOrderCard';

import {OrderList} from '../OrderList/OrderList';

const generateExampleItems = (length) => {
	let exampleItems = [];
	for (let index = 0; index < length; index++) {
		exampleItems.push({
			_id: `item${index}`,
			imageSrc: 'https://example.com',
			rarity: 'legendary'
		});
	}

	return exampleItems;
};

const setup = (propOverrides) => {
	const props = Object.assign({
		items: generateExampleItems(2)
	}, propOverrides);

	const wrapper = shallow(<OrderList {...props}/>);

	return {props, wrapper};
};

describe('Order Component', () => {
	it('renders correctly', () => {
		const {wrapper} = setup();

		expect(wrapper).toMatchSnapshot();
	});

	it('shows an ordered items list from a redux state', () => {
		const {wrapper, props} = setup({
			items: generateExampleItems(11)
		});

		const orderedItemsLength = wrapper.find(ItemOrderCard).length;

		expect(orderedItemsLength).toBe(props.items.length);
	});

	it('shows items with a size depending on an items count', () => {
		const {wrapper} = setup();

		const itemBoxClass = '.order-list--item-box';

		wrapper.setProps({items: generateExampleItems(3)});
		const classesWithLessItems = wrapper.find(itemBoxClass).at(0).props().className;

		wrapper.setProps({items: generateExampleItems(20)});
		const classesWithMoreItems = wrapper.find(itemBoxClass).at(0).props().className;

		expect(classesWithLessItems).not.toBe(classesWithMoreItems);
	});
});