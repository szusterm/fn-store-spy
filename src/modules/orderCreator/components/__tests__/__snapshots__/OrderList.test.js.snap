// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order Component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OrderList
    items={
      Array [
        Object {
          "_id": "item0",
          "imageSrc": "https://example.com",
          "rarity": "legendary",
        },
        Object {
          "_id": "item1",
          "imageSrc": "https://example.com",
          "rarity": "legendary",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container-fluid order-list"
      >
        <div
          className="row"
        >
          <div
            className="col-4 order-list--item-box"
          >
            <Connect(ItemOrderCard)
              imageSrc="https://example.com"
              index={0}
              rarity="legendary"
            />
          </div>
          <div
            className="col-4 order-list--item-box"
          >
            <Connect(ItemOrderCard)
              imageSrc="https://example.com"
              index={1}
              rarity="legendary"
            />
          </div>
        </div>
      </div>,
      "className": "col-12",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
        >
          <div
            className="col-4 order-list--item-box"
          >
            <Connect(ItemOrderCard)
              imageSrc="https://example.com"
              index={0}
              rarity="legendary"
            />
          </div>
          <div
            className="col-4 order-list--item-box"
          >
            <Connect(ItemOrderCard)
              imageSrc="https://example.com"
              index={1}
              rarity="legendary"
            />
          </div>
        </div>,
        "className": "container-fluid order-list",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="col-4 order-list--item-box"
            >
              <Connect(ItemOrderCard)
                imageSrc="https://example.com"
                index={0}
                rarity="legendary"
              />
            </div>,
            <div
              className="col-4 order-list--item-box"
            >
              <Connect(ItemOrderCard)
                imageSrc="https://example.com"
                index={1}
                rarity="legendary"
              />
            </div>,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "item0",
            "nodeType": "host",
            "props": Object {
              "children": <Connect(ItemOrderCard)
                imageSrc="https://example.com"
                index={0}
                rarity="legendary"
              />,
              "className": "col-4 order-list--item-box",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "imageSrc": "https://example.com",
                "index": 0,
                "rarity": "legendary",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "item1",
            "nodeType": "host",
            "props": Object {
              "children": <Connect(ItemOrderCard)
                imageSrc="https://example.com"
                index={1}
                rarity="legendary"
              />,
              "className": "col-4 order-list--item-box",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "imageSrc": "https://example.com",
                "index": 1,
                "rarity": "legendary",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container-fluid order-list"
        >
          <div
            className="row"
          >
            <div
              className="col-4 order-list--item-box"
            >
              <Connect(ItemOrderCard)
                imageSrc="https://example.com"
                index={0}
                rarity="legendary"
              />
            </div>
            <div
              className="col-4 order-list--item-box"
            >
              <Connect(ItemOrderCard)
                imageSrc="https://example.com"
                index={1}
                rarity="legendary"
              />
            </div>
          </div>
        </div>,
        "className": "col-12",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
          >
            <div
              className="col-4 order-list--item-box"
            >
              <Connect(ItemOrderCard)
                imageSrc="https://example.com"
                index={0}
                rarity="legendary"
              />
            </div>
            <div
              className="col-4 order-list--item-box"
            >
              <Connect(ItemOrderCard)
                imageSrc="https://example.com"
                index={1}
                rarity="legendary"
              />
            </div>
          </div>,
          "className": "container-fluid order-list",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="col-4 order-list--item-box"
              >
                <Connect(ItemOrderCard)
                  imageSrc="https://example.com"
                  index={0}
                  rarity="legendary"
                />
              </div>,
              <div
                className="col-4 order-list--item-box"
              >
                <Connect(ItemOrderCard)
                  imageSrc="https://example.com"
                  index={1}
                  rarity="legendary"
                />
              </div>,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "item0",
              "nodeType": "host",
              "props": Object {
                "children": <Connect(ItemOrderCard)
                  imageSrc="https://example.com"
                  index={0}
                  rarity="legendary"
                />,
                "className": "col-4 order-list--item-box",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "imageSrc": "https://example.com",
                  "index": 0,
                  "rarity": "legendary",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "item1",
              "nodeType": "host",
              "props": Object {
                "children": <Connect(ItemOrderCard)
                  imageSrc="https://example.com"
                  index={1}
                  rarity="legendary"
                />,
                "className": "col-4 order-list--item-box",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "imageSrc": "https://example.com",
                  "index": 1,
                  "rarity": "legendary",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
